### PR TITLE
commit trigram.py from lesson 4

### PR DESCRIPTION
**Pull request by NetMic**

### CODE REVIEW

**Comment by NetMic** [comment]
Lesson 5, first homework done.

File: students/Net_Michael/lesson3/mailroom.py [hunk 1]
```diff
@@ -24,124 +20,112 @@ def add_donor():
...
+            thank_one_letter(donor_name, donation)
+        elif response == "4":
+             thank_all_letter(donor_name, donation)
+        else:
+             print("Done")
 
+if __name__ == '__main__':
+    if donation[len(donation)-1] != []:
+        donation += ([],)
+    output_result()
```
Comment (charlesrobison) [inline students/Net_Michael/lesson3/mailroom.py]: Looks good!

File: students/Net_Michael/lesson4/mailroom2.py [hunk 2]
```diff
@@ -1,132 +1,147 @@
...
+
+
+# any of the follwing for choices can be executed for desired output
+
+if __name__ == "__main__":
+    sel_choice = output_entry_value()
+    if sel_choice <= 4:
+        switch_func_dict.get(sel_choice)()
+    else:
+        print("Select an integer between 1 and 4")
```
Comment (charlesrobison) [inline students/Net_Michael/lesson4/mailroom2.py]: This is looking good, but you'll want to be able to add donors and donations as well as produce the report and letters.

File: students/Net_Michael/lesson4/trigram.py [hunk 3]
```diff
@@ -3,48 +3,56 @@
...
 
+if __name__ == "__main__":
+    letter1 = clean_text()
+    print("\n", letter1, "\n")
+    trigrams = set_trigram(ent_letter = letter1, len_word = 10, random_start = "no")
+    print("\n", trigrams, "\n")
+    trigrams_fixed = bld_trigram(ent_letter = letter1, len_word = 10, random_start = "no")
+    print("\n", trigrams_fixed, "\n")
+    trigrams_random = bld_trigram(ent_letter = letter1, len_word = 10, random_start = "yes")
+    print("\n", trigrams_random, "\n")
```
Comment (charlesrobison) [inline students/Net_Michael/lesson4/trigram.py]: Much better!

**Pull request merged**